packages/teams-js/src/public/video.ts

```diff
@@ -216,4 +222,159 @@ export namespace video {
   export function isSupported(): boolean {
     return ensureInitialized(runtime) && runtime.supports.video ? true : false;
   }
+
+  /**
+   * @beta
+   * Namespace to get video frames from a media stream.
+   * When the host supports this capability, developer should call {@link mediaStream.registerForVideoFrame} to get the video frames instead of {@link registerForVideoFrame} to get the video frames, callback of {@link registerForVideoFrame} will be ignored when the host supports this capability.
+   */
+  export namespace mediaStream {
+    /**
+     * @beta
+     * Checks if video.mediaStream capability is supported by the host
+     * @returns boolean to represent whether the video.medisStream capability is supported
+     *
+     * @throws Error if {@linkcode app.initialize} has not successfully completed
+     *
+     */
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && isTextureStreamAvailable() && !!runtime.supports.video?.mediaStream;
+    }
+
+    function isTextureStreamAvailable(): boolean {
+      return (
+        typeof window !== 'undefined' &&
+        !!(window['chrome']?.webview?.getTextureStream && window['chrome']?.webview?.registerTextureStream)
+      );
+    }
+
+    /**
+     * @beta
+     * Video frame data extracted from the media stream. More properties may be added in the future.
+     */
+    export type MediaStreamFrameData = {
+      /**
+       * The video frame from the media stream.
+       */
+      videoFrame: VideoFrame;
+    };
+
+    /**
+     * @beta
+     * Video effect change call back function definition.
+     * The video app should resolve the promise to notify a successfully processed video frame.
+     * The video app should reject the promise to notify a failure.
+     */
+    export type VideoFrameCallback = (receivedVideoFrame: MediaStreamFrameData) => Promise<VideoFrame>;
+
+    type MediaStreamInfo = {
+      streamId: string;
+    };
+
+    /**
+     * @beta
+     * Register to read the video frames from the media stream provided by the host.
+     * @param frameCallback - The callback to invoke when recieve a video frame from the media stream.
+     * @example
+     * ```typescript
```

Comment:
Nit: Might be good to include comment on what this is commented out/saved for

Reply:
This is how you put a code example in the documentation that gets formatted nicely. See here for an example that's already live: https://learn.microsoft.com/en-us/javascript/api/@microsoft/teams-js/search?view=msteams-client-js-latest